tests/unit/s3/test_connection.py

```diff
@@ -48,6 +48,18 @@ def test_switched(self):
         )
 
 
+class TestAnon(MockServiceWithConfigTestCase):
+    connection_class = S3Connection
+
+    def test_generate_url(self):
+        conn = self.connection_class(
+            anon=True,
```

Comment:
Yeah, the existing tests here (since `anon` defaults to `False`):
https://github.com/boto/boto/blob/af045f93d70fbb4cdac2a8e57d040ce59935d45b/tests/unit/s3/test_connection.py#L67-L73
https://github.com/boto/boto/blob/af045f93d70fbb4cdac2a8e57d040ce59935d45b/tests/unit/s3/test_connection.py#L208-L214